test/unit/quantum/results/test_frequencies.py

```diff
@@ -59,44 +64,120 @@ class TestBitflipFrequencies:
-            ({0b00: 0, 0b01: 1, 0b10: 2, 0b11: 3}, 0b00, {0b00: 0, 0b01: 1, 0b10: 2, 0b11: 3}),
-            ({0b00: 0, 0b01: 1, 0b10: 2, 0b11: 3}, 0b01, {0b00: 1, 0b01: 0, 0b10: 3, 0b11: 2}),
-            ({0b00: 0, 0b01: 1, 0b10: 2, 0b11: 3}, 0b10, {0b00: 2, 0b01: 3, 0b10: 0, 0b11: 1}),
-            ({0b00: 0, 0b01: 1, 0b10: 2, 0b11: 3}, 0b11, {0b00: 3, 0b01: 2, 0b10: 1, 0b11: 0}),
+            (Counts, {0b00: 0, 0b01: 1}, 0b00, {0b00: 0, 0b01: 1}),
+            (Counts, {0b00: 0, 0b01: 1}, 0b01, {0b00: 1, 0b01: 0}),
+            (Counts, {0b00: 0, 0b01: 1}, 0b10, {0b10: 0, 0b11: 1}),
+            (Counts, {0b00: 0, 0b01: 1}, 0b11, {0b10: 1, 0b11: 0}),
+            (
+                Counts,
+                {0b00: 0, 0b01: 1, 0b10: 2, 0b11: 3},
+                0b00,
+                {0b00: 0, 0b01: 1, 0b10: 2, 0b11: 3},
+            ),
+            (
+                Counts,
+                {0b00: 0, 0b01: 1, 0b10: 2, 0b11: 3},
+                0b01,
+                {0b00: 1, 0b01: 0, 0b10: 3, 0b11: 2},
+            ),
+            (
+                Counts,
+                {0b00: 0, 0b01: 1, 0b10: 2, 0b11: 3},
+                0b10,
+                {0b00: 2, 0b01: 3, 0b10: 0, 0b11: 1},
+            ),
+            (
+                Counts,
+                {0b00: 0, 0b01: 1, 0b10: 2, 0b11: 3},
+                0b11,
+                {0b00: 3, 0b01: 2, 0b10: 1, 0b11: 0},
+            ),
+            (QuasiDistribution, {0b00: 0, 0b01: 1}, 0b00, {0b00: 0, 0b01: 1}),
+            (QuasiDistribution, {0b00: 0, 0b01: 1}, 0b01, {0b01: 0, 0b00: 1}),
+            (QuasiDistribution, {0b00: 0, 0b01: 1}, 0b10, {0b10: 0, 0b11: 1}),
+            (QuasiDistribution, {0b00: 0, 0b01: 1}, 0b11, {0b11: 0, 0b10: 1}),
+            (
+                QuasiDistribution,
+                {0b00: 0, 0b01: 0.25, 0b10: 0.25, 0b11: 0.5},
+                0b00,
+                {0b00: 0, 0b01: 0.25, 0b10: 0.25, 0b11: 0.5},
+            ),
+            (
+                QuasiDistribution,
+                {0b00: 0, 0b01: 0.25, 0b10: 0.25, 0b11: 0.5},
+                0b01,
+                {0b01: 0, 0b00: 0.25, 0b11: 0.25, 0b10: 0.5},
+            ),
+            (
+                QuasiDistribution,
+                {0b00: 0, 0b01: 0.25, 0b10: 0.25, 0b11: 0.5},
+                0b10,
+                {0b10: 0, 0b11: 0.25, 0b00: 0.25, 0b01: 0.5},
+            ),
+            (
+                QuasiDistribution,
+                {0b00: 0, 0b01: 0.25, 0b10: 0.25, 0b11: 0.5},
+                0b11,
+                {0b11: 0, 0b10: 0.25, 0b01: 0.25, 0b00: 0.5},
+            ),
         ],
     )
-    def test_bitflip_frequencies(self, counts, bitflips, expected):
+    def test_bitflip_frequencies(self, frequency_type, frequencies, bitflips, expected):
         """Test bitflip frequencies base functionality."""
-        counts = Counts(counts)
-        assert bitflip_frequencies(counts, bitflips) == Counts(expected)
+        frequencies = frequency_type(frequencies)
```

Comment:
If you are only going to do this, would it be cleaner to pass the right type from the beginning? 🤔

Reply:
It has the purpose of preventing the initialization on both the inputs and outputs in the parameter set and keeping the code a bit more shallow.